app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditActivity.kt

```diff
@@ -41,19 +43,33 @@ class ProfileEditActivity : InjectableAutoLocalizedAppCompatActivity() {
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    (activityComponent as ActivityComponentImpl).inject(this)
-    profileEditActivityPresenter.handleOnCreate()
-  }
-
-  override fun onBackPressed() {
-    val isMultipane = intent.extras!!.getBoolean(IS_MULTIPANE_EXTRA_KEY, false)
-    if (isMultipane) {
-      @Suppress("DEPRECATION") // TODO(#5404): Migrate to a back pressed dispatcher.
-      super.onBackPressed()
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+      onBackInvokedDispatcher.registerOnBackInvokedCallback(1000) {
+        val isMultipane: Boolean = intent.extras!!.getBoolean(IS_MULTIPANE_EXTRA_KEY, false)
+        if (isMultipane) {
+          onBackPressedDispatcher.onBackPressed()
+        } else {
+          val intent = Intent(this, ProfileListActivity::class.java)
+          intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+          startActivity(intent)
+        }
+      }
     } else {
-      val intent = Intent(this, ProfileListActivity::class.java)
-      intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-      startActivity(intent)
+      onBackPressedDispatcher.addCallback(
+        this@ProfileEditActivity, object: OnBackPressedCallback(true) {
+          override fun handleOnBackPressed() {
+            val isMultipane: Boolean = intent.extras!!.getBoolean(IS_MULTIPANE_EXTRA_KEY, false)
+            if (isMultipane) {
+              onBackPressedDispatcher.onBackPressed()
+            } else {
+              val intent = Intent(this@ProfileEditActivity, ProfileListActivity::class.java)
+              intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+              startActivity(intent)
+            }
+          }
+        })
     }
+    (activityComponent as ActivityComponentImpl).inject(this)
+    profileEditActivityPresenter.handleOnCreate()
   }
```

Comment:
We can improve readability by putting this in a function.
```suggestion
override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    ....
    handleBackPress()
  }

  private fun handleBackPress() {
    onBackPressedDispatcher.addCallback(
      this@AdministratorControlsActivity, object : OnBackPressedCallback(true) {
        override fun handleOnBackPressed() {
          val fragment =
            supportFragmentManager.findFragmentById(
              R.id.administrator_controls_fragment_multipane_placeholder
            )
          /*
           * If the current fragment is ProfileListFragment then the activity should end on back press.
           * If it's instead ProfileEditFragment then profileListFragment should be inflated via
           * handleOnBackPressed.
           */
          if (fragment is ProfileEditFragment) {
            administratorControlsActivityPresenter.handleOnBackPressed()
          } else {
            isEnabled = false
            onBackPressedDispatcher.onBackPressed()
            isEnabled = true
          }
        }
      })
  }
```